include/ck/host_utility/flush_cache.hpp

```diff
@@ -28,19 +28,27 @@ struct RotatingMemWrapperMultiABD
 
     RotatingMemWrapperMultiABD() = delete;
     RotatingMemWrapperMultiABD(Argument& arg_,
-                               std::size_t rotating_count_,
+                               std::size_t rotating_count_hint,
                                std::array<std::size_t, NumAs> size_as_,
                                std::array<std::size_t, NumBs> size_bs_,
                                std::array<std::size_t, NumDs> size_ds_)
         : arg(arg_),
-          rotating_count(rotating_count_),
+          rotating_count(rotating_count_hint),
           size_as(size_as_),
           size_bs(size_bs_),
           size_ds(size_ds_)
     {
         p_as_grids.push_back(arg.p_as_grid);
         p_bs_grids.push_back(arg.p_bs_grid);
         p_ds_grids.push_back(arg.p_ds_grid);
+
+        // limit the rotating count to prevent oom
+        const uint64_t footprint = std::accumulate(size_as.begin(), size_as.end(), 0UL) +
+                                   std::accumulate(size_bs.begin(), size_bs.end(), 0UL) +
+                                   std::accumulate(size_ds.begin(), size_ds.end(), 0UL);
+        const uint64_t max_rotating_count = (1ULL << 31) / footprint;
```

Comment:
2GB buffer here seems somewhat arbitrary, given the number of devices we need to support. What if we checked [hipMemGetInfo](https://rocm.docs.amd.com/projects/HIP/en/docs-develop/doxygen/html/group___memory.html#ga311c3e246a21590de14478b8bd063be2) and use a portion of that (not sure how many processes we expect to run concurrently with this feature..)? Then we may not have to limit ourselves to 2GB problem sizes.

Reply:
Ok, makes sense